### ISSUE TITLE
`reportUninitializedInstanceVariable`: in final subclasses of ABC, report also uninitialized variables from the parents

### ISSUE THREAD

**Issue by tmke8** [microsoft/pyright#5103]
**Is your feature request related to a problem? Please describe.**

Recently, pyright gained this new behavior, which is great:
```python
from abc import ABC
from typing import final

class E(ABC):
    x: str

@final
class ESub(E):
    y: str  # E: Instance variable "y" is not initialized in the class body or __init__ method
```
However, `x` is also uninitialized in `ESub`.

**Describe the solution you'd like**
Ideally, I would want pyright to also complain about the `x` which is also uninitialized in this case.

That is, I'm arguing for this behavior:
```python
from abc import ABC
from typing import final

class E(ABC):
    x: str

@final
class ESub(E):  # E: Instance variable "x" is not initialized in the class body or __init__ method
    pass
```
Basically, pyright keeps track of which variables it *would have* complained about in the abstract classes, and then complains about them in all the final subclasses.

**Comment by erictraut** [comment]
Pyright already includes similar functionality to what you're requesting. It complains about members that are defined in a base _protocol class_ that are not implemented in a derived class. 

Is there a reason you are using `ABC` rather than `Protocol` for your base class? The type checking rules for protocols are well defined (in [PEP 544](https://peps.python.org/pep-0544/)) and enabled by default even in "basic" type checking mode. The rules for `reportUninitializedInstanceVariable` are much more nebulous because this check is enforcing a set of coding conventions that are not used in most Python code bases. (That's why it is not enabled by default even in "strict" mode.)

**Comment by tmke8** [comment]
> Is there a reason you are using `ABC` rather than `Protocol` for your base class?

Mostly because `isinstance` is more reliable with `ABC`s. But yes, my alternative is using `Protocol`s.

**Comment by adampauls** [comment]
+1, Protocols are not a replacement for ABCs if you want nominal typing. If this is an easy add, I think it would be great!

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.309, which I just published. It will also be included in a future release of pylance.
